docs/design/2022-10-13-mysql-compatible-auto-increment.md

```diff
@@ -0,0 +1,109 @@
+# Proposal: MySQL compatible `AUTO_INCREMENT`
+
+- Author(s): [tiancaiamao](https://github.com/tiancaiamao)
+- Discussion PR: https://github.com/pingcap/tidb/pull/38449
+- Tracking Issue: https://github.com/pingcap/tidb/issues/38442
+
+## Abstract
+
+This proposes an implementation of making the `AUTO_INCREMENT` behaviour compatible with MySQL.
+
+## Background
+
+MySQL deploys on a single-machine, and it provides the [`AUTO_INCREMENT`](https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html) table attribute to generate a unique identity for new rows.
+
+In TiDB, we support generating `AUTO_INCREMENT` IDs that are **unique**, **monotone increasing**, but the IDs may **not be consecutive**.
+The current behaviour is not fully-compatible with MySQL.
```

Comment:
`The current behaviour in TiDB is not fully-compatible with MySQL.` is better

And maybe new line?

Reply:
Done